### PR TITLE
fix: Fix handling empty market data

### PR DESCRIPTION
**Pull request by hiohiohio**
fixes https://github.com/alpacahq/alpaca-py/issues/317

Context:
- we currently cannot handle empty response when querying trades/quotes/snapshot/etc...
- this PR fixes unintended errors when handling empty response.

Changes:
- returning empty object/dict instead of raising an error when API returns empty data

### CODE REVIEW

File: tests/data/test_historical_stock_data.py [hunk 1]
```diff
@@ -777,3 +1123,49 @@ def test_multi_stock_latest_bar(reqmock, stock_client: StockHistoricalDataClient
...
+
+def test_stock_latest_bar_single_empty_response(
+    reqmock, stock_client: StockHistoricalDataClient
+):
+    symbol = "SPY"
+    reqmock.get(
+        f"https://data.alpaca.markets/v2/stocks/{symbol}/bars/latest",
+        text="""
+        {
+            "message":"no bar found for OTOC"
```
Comment (hiohiohio) [inline tests/data/test_historical_stock_data.py:1136]: this should be 
```
        "next_page_token": null,
         "bars": null,
         "symbol": "AAPL"
```

File: tests/data/test_historical_stock_data.py [hunk 2]
```diff
@@ -695,6 +995,52 @@ def test_get_snapshot(reqmock, stock_client: StockHistoricalDataClient):
...
+def test_get_snapshot_single_empty_response(
+    reqmock, stock_client: StockHistoricalDataClient
+):
+    symbol = "AAPL"
+
+    reqmock.get(
+        f"https://data.alpaca.markets/v2/stocks/{symbol}/snapshot",
+        text="""
+    {
+        "message":"no snapshot found for OTOC"
```
Comment (hiohiohio) [inline tests/data/test_historical_stock_data.py:1007]: should be different but need to find actual empty response from API

File: tests/data/test_historical_stock_data.py [hunk 3]
```diff
@@ -609,6 +860,55 @@ def test_get_latest_quote(reqmock, stock_client: StockHistoricalDataClient):
...
+    reqmock, stock_client: StockHistoricalDataClient
+):
+    # Test single symbol request
+    symbol = "AAPL"
+
+    reqmock.get(
+        f"https://data.alpaca.markets/v2/stocks/{symbol}/quotes/latest",
+        text="""
+    {
+        "message":"no quote found for AAPL"
```
Comment (hiohiohio) [inline tests/data/test_historical_stock_data.py:873]: should be
```
        "next_page_token": null,
         "quotes": null,
         "symbol": "AAPL"
```

File: alpaca/data/models/bars.py [hunk 4]
```diff
@@ -71,7 +76,10 @@ def __init__(
 
         raw_bars = raw_data
 
-        for symbol, bars in raw_bars.items():
-            parsed_bars[symbol] = [Bar(symbol, bar) for bar in bars]
+        if raw_bars is not None:
+            for symbol, bars in raw_bars.items():
+                parsed_bars[symbol] = [
+                    Bar(symbol, bar) for bar in bars if bar is not None
```
Comment (matebudai) [inline alpaca/data/models/bars.py:82]: Not important, but interestingly pylint complains about our naming, it black listed `foo, bar` words.
`bars.py:82:41: C0104: Disallowed name "bar" (disallowed-name)`

**Review by matebudai — APPROVED** [review]
We have a few unused imports here and there, like List, Dict, etc. nothing import though.
LGTM.

**Pull request merged**